modules/heat_conduction/doc/content/source/constraints/InterfaceJouleHeatingConstraint.md

```diff
@@ -0,0 +1,82 @@
+# InterfaceJouleHeatingConstraint
+
+!syntax description /Constraints/InterfaceJouleHeatingConstraint
+
+## Description
+
+The `InterfaceJouleHeatingConstraint` class is intended to calculate add the heat source due to Joule Heating which results from the electric potential drop across an interface.
+The heat source is then added to the temperature field variable  in a coupled electro-thermal simulation.
+This class is intended to be used in conjunction with [ModularGapConductanceConstraint](ModularGapConductanceConstraint.md) and [GapFluxModelPressureDependentConduction](GapFluxModelPressureDependentConduction.md), which enforce the closed gap interface requirement by checking for a postive normal pressure.
+As such, the `InterfaceJouleHeatingConstraint` takes as a required argument the name of the Lagrange Multiplier variable used in the electrical contact.
+
+!alert note title=Closed Gap Interface Assumed by this Class
+The `InterfaceJouleHeatingConstraint` class should only be employed when the user is certain that electric potential flux is nonzero only when the interface gap is closed.
```

Comment:
Can the object still be used in a transient simulation when these conditions do not take place until a while into the simulation?

Reply:
I've tested that the class behaves as expected (no electrical flux across an open gap and no resulting interface heating) and have updated the documentation to reflect that behavior. The test I currently have is much to long running as a regression test (to show starting out of contact and coming into contact), so I am currently working on developing a regression style test for this condition

Reply:
New test, with some portion of open and closed gaps along the interface, added